3212harsh/Exploit_Arena
# implement adding new problem and authentication

Changes Done

1. Authentication Using JWT:
- Implemented authentication using JWT (JSON Web Tokens) for securing routes and ensuring that only authenticated users can perform specific actions.
- Created a new middleware file auth.middleware.js to handle JWT verification and authentication logic for protected routes.
- The middleware checks for a valid JWT token in the request header and ensures the user is authenticated before granting access to certain resources.

2. Changes to `user` and `problem` Models:
- User Model:
  - Made necessary updates to the `user` model to support the new authentication mechanism. These updates include adding and managing user data relevant to authentication.
- Problem Model:
  - Modified the `problem` model to accommodate new features related to adding a problem. These changes allow the system to handle problem-related data more effectively.

3. Backend for Adding a New Problem:
- Developed the backend functionality for adding a new problem to the system. 
- This includes creating a route for adding problems, processing incoming requests, and storing problem data in the database.
- The feature allows authenticated users to submit new problems through a simple API endpoint.

How to Test:
1. Ensure that the JWT authentication is properly set up and the auth.middleware.js file is correctly integrated into the routes.
2. Test adding a new problem via the created API endpoint. Ensure that only authenticated users can successfully add a new problem.
3. Validate that the problem data is correctly stored in the database.

Note:- Please ensure that the JWT token is included in the request header when testing protected routes.
